Fix xcframework import when framework name is different from xcframework name

AlbGarciam opened the pull request:
### Short description 📝

The reason behind using framework's name instead xcframework's name is because we can have a xcframework with a different name than the library that is compiled on it. 

For example, I can have 2 xcframeworks NaluPlayback and NaluPlaybackLite, both will have their product's name as NaluPlayback in order to import both of them in the same way `import NaluPlayback` but when generating the xcframework, as it uses the scheme's name to create the xcframework's name they are going to be deployed as NaluPlaybackLite.xcframework and NaluPlayback.xcframework. But the linked libraries inside both xcframework will be NaluPlayback.framework

If we use the current solution, we will have an error like:
```
NaluPlaybackLite.xcframework is missing architecture ios-arm64/NaluPlayback.framework/NaluPlaybackLite defined in the Info.plist
```

### How to test the changes locally 🧐

If you change the xcframework's name without updating the library it should fail

### Checklist ✅

- [X] The code architecture and patterns are consistent with the rest of the codebase.
- [X] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [X] In case the PR introduces changes that affect users, the documentation has been updated.
- [X] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

danyf90 reviewed (changes requested):
Thanks for the PR! 🙌 

Could you please also:
- add a unit test in `XCFrameworkMetadataProviderTests`
- change one of the framework in the `projects/tuist/fixtures/ios_app_with_xcframeworks` to have a library name different from the framework name, so that it is tested also in the acceptance tests
- Use proper PR title and labels so we can check `The PR includes the label changelog:added, changelog:fixed, or changelog:changed, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.`

🚀

danyf90 commented:
@allcontributors add @AlbGarciam for code

kwridan commented on Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift:
```diff
@@ -128,7 +128,7 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
             binaryPath = AbsolutePath(library.identifier, relativeTo: xcframeworkPath)
                 .appending(RelativePath(library.path.pathString))
-                .appending(component: binaryName)
+                .appending(component: library.path.pathString.replacingOccurrences(of: ".framework", with: ""))
```
This may not achieve the desired results as `pathString` is a path rather than a file name. 

I believe `.basenameWithoutExt` may be more appropriate.

AlbGarciam replied:
You were right. Thanks!

kwridan reviewed (commented):
Thanks for contributing this @AlbGarciam

AlbGarciam commented:
> Thanks for the PR! 🙌
> 
> Could you please also:
> 
> * add a unit test in `XCFrameworkMetadataProviderTests`
> * change one of the framework in the `projects/tuist/fixtures/ios_app_with_xcframeworks` to have a library name different from the framework name, so that it is tested also in the acceptance tests
> * Use proper PR title and labels so we can check `The PR includes the label changelog:added, changelog:fixed, or changelog:changed, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.`
> 
> 🚀

I have added a new framework on `Tests/Fixtures/` called `MyFrameworkDifferentProductName.xcframework`which basically has the structure that I described on the ticket and I added the tests accordingly.

AlbGarciam commented:
@danyf90 regarding adding the proper label to the PR I can't add any label to it.

kwridan reviewed (approved):
Thanks for the updates @AlbGarciam

AlbGarciam commented:
@danyf90 Do you know when is the next release planned?
